### PR TITLE
k-points for the GW self-energy

### PR DESCRIPTION
**Pull request by JWilhelm**
Enables computation of band structures from GW; yet critical to do: stabilize calculations for large RI basis sets

### CODE REVIEW

File: src/hfx_ri.F [hunk 1]
```diff
@@ -1560,6 +1560,8 @@ SUBROUTINE hfx_ri_update_ks_Pmat(qs_env, ri_data, ks_matrix, rho_ao, &
          CALL dbt_batched_contract_finalize(ri_data%t_3c_int_ctr_1(1, 1))
          CALL dbt_batched_contract_finalize(t_3c_3)
 
+         PRINT *, 'n_mem =', n_mem, 'n_mem_RI =', n_mem_RI
```
Comment (fstein93) [inline src/hfx_ri.F]: Please remove this line.

File: src/rpa_gw.F [hunk 2]
```diff
@@ -1451,6 +1498,38 @@ SUBROUTINE compute_QP_energies(vec_Sigma_c_gw, count_ev_sc_GW, gw_corr_lev_occ,
 
    END SUBROUTINE compute_QP_energies
 
+! **************************************************************************************************
+!> \brief ...
+!> \param vec_gw_energ ...
+!> \param z_value ...
+!> \param m_value ...
+! **************************************************************************************************
+   SUBROUTINE check_NaN(vec_gw_energ, z_value, m_value)
```
Comment (fstein93) [inline src/rpa_gw.F]: What do you think about the following suggestion:
The routine checks only a single array (ALLOCATABLE attribute is not necessary here) and sets NaNs to a user-given value.

File: src/hfx_ri.F (hunk 1 above)
Comment (JWilhelm) [reply src/hfx_ri.F]: Thanks for spotting this, it's gone now.

File: src/rpa_gw.F (hunk 2 above)
Comment (JWilhelm) [reply src/rpa_gw.F]: Yes, good point, we can certainly do, code is updated.

**Comment by JWilhelm** [comment]
In case additional code modifications are benefitial to get the PR accepted, I will be happy to make them.

**Pull request merged**